returnn/tf/layers/basic.py

```diff
@@ -2945,6 +2945,49 @@ def get_out_data_from_opts(cls, name, axis, repetitions, sources=(), **kwargs):
     return data
 
 
+class TileLayer(_ConcatInputLayer):
+  """
+  A wrapper around tf.tile
+  """
+  layer_class = "tile"
+
+  def __init__(self, multiples, **kwargs):
+    """
+    :param dict[str, int] multiples: number of multiples per axis (axis provided as str)
+    """
+    from returnn.tf.util.data import DimensionTag
+    super(TileLayer, self).__init__(**kwargs)
+    self.multiples = multiples
+    input_data = self.input_data
+
+    multiples_full = [1] * input_data.batch_ndim
+    for axis, multiple in multiples.items():
+      a = input_data.get_axis_from_description(axis)
+      assert a not in input_data.get_dynamic_axes(), "Tiling of dynamic axes not yet implemented"
+      assert a != input_data.batch_dim_axis, "Tiling of batch axis not yet implemented"
+      multiples_full[a] *= multiple
+
+    self.output.placeholder = tf.tile(input_data.placeholder, tf.constant(multiples_full))
+
+  @classmethod
+  def get_out_data_from_opts(cls, name, multiples, sources=(), **kwargs):
+    """
+    :param str name:
+    :param dict[str, int] multiples:
+    :param list[LayerBase] sources:
+    :rtype: Data
+    """
+    data = get_concat_sources_data_template(sources, name="%s_output" % name)
+    for axis, multiple in multiples.items():
+      a = data.get_batch_axis_excluding_batch(data.get_axis_from_description(axis))
+      if a is None or data.shape[a] is None or a + 1 in data.get_dynamic_axes():
```

Comment:
Why `a + 1`? This is wrong in general.
But also, this is just the same as `data.shape[a] is None`. So you don't need that.
```suggestion
      a = data.get_batch_axis_excluding_batch(data.get_axis_from_description(axis, allow_int=False))
      if a is None or data.shape[a] is None:
```